src/Microsoft.Azure.WebJobs.Host/InvokeFunctionFilterAttribute.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    /// <summary>
+    /// An invocation filter that invokes job methods
+    /// </summary>
+    [CLSCompliant(false)]
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, Inherited = true, AllowMultiple = true)]
+    public sealed class InvokeFunctionFilterAttribute : InvocationFilterAttribute
+    {
+        /// <summary>
+        /// When this attribute is used, take the name of the function to invoke
+        /// </summary>
+        /// <param name="executingFilter"></param>
+        /// <param name="executedFilter"></param>
+        public InvokeFunctionFilterAttribute(string executingFilter = null, string executedFilter = null)
+        {
+            ExecutingFilter = executingFilter;
+            ExecutedFilter = executedFilter;
+        }
+
+        /// <summary>
+        /// Executing filter
+        /// </summary>
+        public string ExecutingFilter { get; }
+
+        /// <summary>
+        /// Executed filter
+        /// </summary>
+        public string ExecutedFilter { get; }
+
+        /// <summary>
+        /// Call the requested job method before the main function call
+        /// </summary>
+        /// <returns></returns>
+        public override async Task OnExecutingAsync(FunctionExecutingContext executingContext, CancellationToken cancellationToken)
+        {
+            if (executingContext == null)
+            {
+                throw new ArgumentNullException("executingContext");
+            }
+            
+            if (!string.IsNullOrEmpty(ExecutingFilter))
+            {
+                executingContext.Logger.LogInformation("Executing Function Filter '" + ExecutingFilter + "'");
+
+                try
+                {
+                    await InvokeJobFunctionAsync(executingContext, cancellationToken);
+                }
+                catch (Exception e)
+                {
```

Comment:
We shouldn't be swallowing errors. All of the error paths for filter invocations need to be reviewed. I think the behavior should be:

- if any executing filters throw, skip the rest of the executing filters, skip the user method, and invoke executing filters
- if any executed filter throws, skip the rest of the executed filters